src/version-util.ts

```diff
@@ -18,35 +18,83 @@
  * Contains version utility functions.
  */
 
-import * as httpm from 'typed-rest-client/HttpClient';
+import https from 'https';
+import { URL } from 'url';
 import { retry } from '@lifeomic/attempt';
 
+import { userAgentString } from './user-agent-util';
+
+// cloudSDKComponentsURL is the URL where the JSON file that lists Cloud SDK
+// versions resides.
+const cloudSDKComponentsURL = 'https://dl.google.com/dl/cloudsdk/channels/rapid/components-2.json';
+
 /**
+ * getLatestGcloudSDKVersion fetches the latest version number from the API.
+ *
  * @returns The latest stable version of the gcloud SDK.
  */
 export async function getLatestGcloudSDKVersion(): Promise<string> {
-  const queryUrl = 'https://dl.google.com/dl/cloudsdk/channels/rapid/components-2.json';
-  const client: httpm.HttpClient = new httpm.HttpClient('github-actions-setup-gcloud-sdk');
-  return await retry(
-    async () => {
-      const res = await client.get(queryUrl);
-      if (res.message.statusCode != 200) {
-        throw new Error(
-          `Failed to retrieve gcloud SDK version, HTTP error code: ${res.message.statusCode} url: ${queryUrl}`,
-        );
-      }
-
-      const body = await res.readBody();
-      const responseObject = JSON.parse(body);
-      if (!responseObject.version) {
-        throw new Error(`Failed to retrieve gcloud SDK version, invalid response body: ${body}`);
-      }
-      return responseObject.version;
-    },
-    {
-      delay: 200,
-      factor: 2,
-      maxAttempts: 4,
+  const retryOpts = {
+    delay: 200,
+    factor: 2,
+    maxAttempts: 3,
+  };
+
+  return await retry(async () => {
+    return await getGcloudVersion(cloudSDKComponentsURL);
+  }, retryOpts);
+}
+
+/**
+ * getGcloudVersion is a helper that actually downloads the gcloud version
+ * information.
+ *
+ * @param url URL for the component download. This is exposed for testing, but
+ * should be called without parameters.
+ *
+ */
+async function getGcloudVersion(url: string): Promise<string> {
```

Comment:
I had aspirations of writing unit tests for this, but spinning up an http(s) server for testing is surprisingly hard in node.